packages/analyticsdx-template-lint/src/linter.ts

```diff
@@ -66,17 +66,26 @@ function findAllItemsForLayoutDefinition(layoutJson: JsonNode): JsonNode[] {
  */
 function findAllVariableNamesForLayoutDefinition(layoutJson: JsonNode): Array<{ name: string; nameNode: JsonNode }> {
   return findAllItemsForLayoutDefinition(layoutJson).reduce((items, item) => {
-    const type = findJsonPrimitiveAttributeValue(item, 'type')[0];
-    if (type === 'Variable') {
-      const [name, nameNode] = findJsonPrimitiveAttributeValue(item, 'name');
-      if (typeof name === 'string' && nameNode) {
-        items.push({ name, nameNode });
-      }
-    }
-    return items;
+    const variableItems = findAllVariableItemsForLayoutDefinition(item);
+    return [...items, ...variableItems];
   }, [] as Array<{ name: string; nameNode: JsonNode }>);
 }
 
+function findAllVariableItemsForLayoutDefinition(item: JsonNode): Array<{ name: string; nameNode: JsonNode }> {
```

Comment:
Nit: `findAllVariableItemsForLayoutItem` is probably a better name (since it's an item being passed in, not a LayoutDefinition).